### DISCUSSION TITLE
YetiforcePDF and PHP 7.4

### DISCUSSION THREAD

**Discussion by IgorRSU** [YetiForceCompany/YetiForceCRM#15023] [General]
Hi, I'm using YF 6.2 stable running on PHP 7.4 and I have a problem with generating PDF - if Template is small - I get broken PDF file, if large - I get nothing.
Error log:
`PHP message: PHP Warning:  bcsub(): bcmath function argument is not well-formed`
 in *vendor/yetiforce/yetiforcepdf/lib/Math.php on line 90

I can't switch php to 7.3. Any ideas?

**Comment by mariuszkrzaczkowski** [comment]
Have you checked on https://gitdeveloper.yetiforce.com/?

**Reply by IgorRSU** [reply]
Sales invoice is generating Ok, but I can't access to admin panel on gitdeveloper.
In my YF generating Sales invoice do noting

**Reply by mariuszkrzaczkowski** [reply]
admin/admin 
https://gitdeveloper.yetiforce.com/

**Reply by IgorRSU** [reply]
Thank You,
I have imported my PDF template to GitDev and it works good, in my YF I get broken PDF file

**Reply by mariuszkrzaczkowski** [reply]
So it's not a PHP version but something else,
run full logs and look for something suspicious

**Reply by IgorRSU** [reply]
Im looking in my server Error Log (I'm using nazwa pl) and I have only lot of the same errors:
`PHP message: PHP Warning: bcsub(): bcmath function argument is not well-formed in */vendor/yetiforce/yetiforcepdf/lib/Math.php on line 90`

**Reply by mariuszkrzaczkowski** [reply]
I think this is a bug
https://www.php.net/manual/en/migration74.incompatible.php#migration74.incompatible.bcmath:

**Reply by IgorRSU** [reply]
As far as I understand - this bug has no effect on my problem with generating broken PDF ?

**Reply by mariuszkrzaczkowski** [reply]
I don't know, it requires verification
